presto-main/src/main/java/com/facebook/presto/connector/informationSchema/InformationSchemaPageSourceProvider.java

```diff
@@ -204,6 +210,32 @@ private InternalTable buildTables(Session session, String catalogName, Map<Strin
         return metadata.listTables(session, extractQualifiedTablePrefix(catalogName, filters));
     }
 
+    private InternalTable buildTablePrivileges(Session session, String catalogName, Map<String, NullableValue> filters)
+    {
+        List<GrantInfo> grants = ImmutableList.copyOf(getTablePrivilegesList(session, catalogName, filters));
+
+        InternalTable.Builder table = InternalTable.builder(informationSchemaTableColumns(TABLE_TABLE_PRIVILEGES));
+        for (GrantInfo grant : grants) {
+            if (session.getIdentity().getUser().equals(grant.getIdentity().getUser())) {
+                for (PrivilegeInfo privilegeInfo : grant.getPrivilegeInfo()) {
+                    table.add(
+                            grant.getIdentity().getUser(),
+                            catalogName,
+                            grant.getSchemaTableName().getSchemaName(),
+                            grant.getSchemaTableName().getTableName(),
+                            privilegeInfo.getPrivilege().name(),
+                            privilegeInfo.isGrantOption());
+                }
+            }
+        }
+        return table.build();
+    }
+
+    private List<GrantInfo> getTablePrivilegesList(Session session, String catalogName, Map<String, NullableValue> filters)
+    {
+        return metadata.listTablePrivileges(session, extractQualifiedTablePrefix(catalogName, filters), extractGrantee(session, catalogName, filters));
```

Comment:
As of now, it will throw an exception for all catalogs except Hive. Not sure what information to show. 
Regrading first comment: yeah, that should be taken care of.